runtime/jvmti/jvmtiStackFrame.c

```diff
@@ -283,16 +283,27 @@ jvmtiGetFrameCount(jvmtiEnv* env,
 		rc = getVMThread(currentThread, thread, &targetThread, TRUE, TRUE);
 		if (rc == JVMTI_ERROR_NONE) {
 			J9StackWalkState walkState;
-
-			vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
-
-			walkState.walkThread = targetThread;
 			walkState.flags = J9_STACKWALK_INCLUDE_NATIVES | J9_STACKWALK_VISIBLE_ONLY;
 			walkState.skipCount = 0;
-			vm->walkStackFrames(currentThread, &walkState);
+
+#if JAVA_SPEC_VERSION >= 19
+			if ((NULL == targetThread)) {
+				j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
+				if (IS_VIRTUAL_THREAD(currentThread, threadObject)) {
```

Comment:
Please handle the `else` case with respect to this` if {` block. i.e. `IS_VIRTUAL_THREAD()` is FALSE. If it is not reachable, please add `Assert_JVMTI_true(0);` in that case.